enhancements/hypershift/node-lifecycle.md

```diff
@@ -0,0 +1,198 @@
+- InPlace:
+  - When a new release version is specified in a NodePool, this results in a new userdata created and updated in the MachineSet which enables upcoming Machines to come up with the targeted version. Existing Machines are upgrade in place (See section below).
+- Scaling is delegated to MachineDeployment/MachineSet by reconciling `.replicas`.
+- Autoscaling is delegated to the [Cluster API Autoscaler provider] (https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi) by reconciling `.NodePoolAutoScaling` into MachineDeployment/MachineSet.
+- AutoRepair is delegated to CAPI by reconciling to create or updated opinionated MachineHealthCheck CRs.
+- Bootstrapping CSR are handle by https://github.com/openshift/cluster-machine-approver.
+
+### OS / kubelet and config management
+In standalone OCP OS and config management is driven by the [MCO](https://github.com/openshift/machine-config-operator).
+The HyperShift form factor prevents their original design from being suitable:
+- ClusterScoped resources and single controller does not fit well in the single management cluster multi-tenancy model.
+- Machine Config Server does not support authentication.
+
+To overcome these limitations while reusing as much as possible of the existing MCO knowledge and learnings, some components are introduced:
+- [Ignition Server](https://github.com/openshift/hypershift/tree/main/ignition-server).
+- [Inplace Upgrader](https://github.com/openshift/hypershift/tree/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader).
+- `MachineConfig` CRs can be embedded in ConfigMaps and associated to NodePools [via the `.config` the API know](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/nodepool_types.go#L118-L127).
+
+#### Ignition Server
+At a high level the Ignition Server is an HTTP request multiplexer.
+It serves Ignition payloads over the /ignition endpoint for a particular "Bearer $token" passed through the "Authorization" Header.
+
+It runs a "token" Secret controller underneath.
+This is the controller that generates and caches the Ignition payload for any pair of `.release` and `.config` input for each NodePool.
+The TokenSecret controller watches "token" Secrets created by the NodePool controller and:
+- Generates the Ignition payload for that `.release` and `.config` pair (See below for details).
+- Stores the payload in memory indexed by "token" and exposes it to the Server.
+- Sets a TTL for the token.
+- After TTL/2 move the original token to rotated spot and create a new one. This results in both tokens for the same `.release` and `.config` pair coexisting during a TTL/2 duration. This give in flight operations e.g. Host booting with an userdata pointing to the rotated token.
```

Comment:
s/This give in flight/This gives in flight/
also seems that the sentence is not finished. I assume it was intended to say something like:
"This gives in flight operations (e.g. Host booting with an userdata pointing to the rotated token) time to complete."